src/Type/Infer.hs

```diff
@@ -1009,10 +1010,10 @@ inferExpr propagated expect (Inject label expr behind rng)
                  -- general handled effects use "@inject-effect"
                  Just coreHTag
                    -> do (maskQName,maskTp,maskInfo) <- resolveFunName nameMaskAt (CtxFunArgs 3 [] Nothing) rng rng
-                         (evvIndexQName,evvIndexTp,evvIndexInfo) <- resolveFunName nameEvvIndex (CtxFunArgs 1 [] Nothing) rng rng
+                         (evvIndexQName,evvIndexTp,evvIndexInfo) <- resolveFunName nameEvvIndex (CtxFunArgs 2 [] Nothing) rng rng
                          let coreMask = coreExprFromNameInfo maskQName maskInfo
                              coreIndex= Core.App (Core.TypeApp (coreExprFromNameInfo evvIndexQName evvIndexInfo) [effTo])
-                                                 [coreHTag]
+                                                 [(makeInt32 0), coreHTag]
```

Comment:
Not sure about this one, this is in `inject-effect`. I assumed the new effect will be the first one of that tag, but I'm not sure if I understand the inject code that well currently.